packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -121,21 +121,35 @@ abstract class LambdaAuthorizer extends Authorizer implements IAuthorizer {
    * Sets up the permissions necessary for the API Gateway service to invoke the Lambda function.
    */
   protected setupPermissions() {
-    if (!this.role) {
-      this.handler.addPermission(`${Names.uniqueId(this)}:Permissions`, {
-        principal: new iam.ServicePrincipal('apigateway.amazonaws.com'),
-        sourceArn: this.authorizerArn,
-      });
-    } else if (this.role instanceof iam.Role) { // i.e. not imported
-      this.role.attachInlinePolicy(new iam.Policy(this, 'authorizerInvokePolicy', {
-        statements: [
-          new iam.PolicyStatement({
-            resources: this.handler.resourceArnsForGrantInvoke,
-            actions: ['lambda:InvokeFunction'],
-          }),
-        ],
-      }));
+    if (this.role instanceof iam.Role) {
+      this.addLambdaInvokePermission(this.role);
+      return;
     }
+    this.addDefaultPermisionRole();
```

Comment:
```suggestion
    if (!this.role) {
        this.addDefaultPermisionRole();
    } else if (this.role instanceof iam.Role) {
      this.addLambdaInvokePermission(this.role);
    }
```

Reply:
@comcalvi 

It sure looks a lot like the previous state of the code, and it's easy to see that no changes have been made!
But I think it would be easier to see if I reduced the unnecessary nesting (else if) in the code, is my original code not so good from your point of view?

Reply:
Ah. I understand.
So this.role is allowed iam.IRole, so there could be side effects if, for example, fromLookup references an existing IAM role?

I would like to incorporate your SUGGESTION.